package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -0,0 +1,519 @@
+        # pulls first key out of 'particles'
+        # allows for arbitrary name of group1 in 'particles'
+        self._particle_group = self._file['particles'][
+                               list(self._file['particles'])[0]]
+
+    def close(self):
+        """close reader"""
+        self._file.close()
+        ## need to change this so it doesnt close the file, since
+        ## file might not be reopened in the same away
+
+    @property
+    def has_positions(self):
+        """True if 'position' group is in trajectory."""
+        return self._has['position']
+
+    @has_positions.setter
+    def has_positions(self, value: bool):
+        self._has['position'] = value
+
+    @property
+    def has_velocities(self):
+        """True if 'velocity' group is in trajectory."""
+        return self._has['velocity']
+
+    @has_velocities.setter
+    def has_velocities(self, value: bool):
+        self._has['velocity'] = value
+
+    @property
+    def has_forces(self):
+        """True if 'force' group is in trajectory."""
+        return self._has['force']
+
+    @has_forces.setter
+    def has_forces(self, value: bool):
+        self._has['force'] = value
+
+    @property
+    def n_frames(self):
+        """number of frames in trajectory"""
+        for name, value in self._has.items():
+            if value:
+                return self._particle_group[name]['value'].shape[0]
+
+    def _reopen(self):
+        """reopen trajectory"""
+        self.close()
+        self.open_trajectory()
+
+    def _read_frame(self, frame):
+        """reads data from h5md file and copies to current timestep"""
+        try:
+            for name, value in self._has.items():
+                if value:
+                    myframe = self._particle_group[name]['step'][frame]
+                    break
+            else:
+                raise NoDataError("Provide at least a position, velocity"
+                                  " or force group in the h5md file.")
+        except ValueError:
+            raise IOError from None
+
+        self._frame = frame
+        ts = self.ts
+        particle_group = self._particle_group
+        ts.frame = frame
+
+        # this block populates the data dictionary
+        # DT not read
+        if 'observables' in self._file:
+            data = self._file['observables']
+            for name in self._data_keywords:
+                self._copy_data(name, data)
+        for name, value in self._has.items():
+            if value:
+                self.ts.data['time'] = particle_group[name]['time'][frame]
+                break
+
+        # set frame box dimensions
+        # set triclinic box vectors
+        ts._unitcell[:] = particle_group['box/edges/value'][frame, :]
+
+        # set the timestep positions, velocities, and forces with
+        # set_ts_attribute() method
+        ts.positions = self._set_ts_attribute('position')
+        ts.velocities = self._set_ts_attribute('velocity')
```

Comment:
This assigns `None` to `ts.velocities` if there are no velocities. It will create `ts.velocities` in the process. I don’t think that the `Timestep` should contain `velocities` if there are none defined. 

You might have to include the 
```python
If self._has[“velocities”]:
    setattr(ts, “velocities”, value)
```
In your method and really make it a true “set” method.